lib/qb_backend/posts/bookmark.ex

```diff
@@ -0,0 +1,60 @@
+defmodule QbBackend.Posts.Bookmark do
+  @moduledoc """
+  This module defines the schema for the bookmark section
+  """
+  use QbBackend.Utils.Schema
+
+  alias QbBackend.{
+    Posts.Manual,
+    Accounts.Profile,
+    Posts.Bookmark
+  }
+
+  schema "bookmarks" do
+    has_many(:manuals, Manual, on_replace: :delete)
+    belongs_to(:profile, Profile, foreign_key: :profile_id, type: :binary_id)
+
+    timestamps(inserted_at: :added_on, updated_at: :updated_on)
+  end
+
+  @doc """
+  This changeset function creates a bookmark
+  """
+  @spec changeset(Bookmark.t(), map) :: {:ok, Bookmark.t()}|{:error, Ecto.Changeset.t()}
+  def changeset(%Bookmark{} = bookmark, params) do
+    bookmark
+    |> cast(params, [])
+  end
+
+  @doc """
+  This function takes a user profile and a manual and adds the bookmark
+  """
+  @spec create_bookmark_with_profile(Profile.t(), map) :: {:ok, Bookmark.t()} | {:error, Ecto.Changeset}
+  def create_bookmark_with_profile(%Profile{} = profile, params) do
+    %Bookmark{}
+    |> changeset(params)
+    |> put_assoc(:profile, profile)
+  end
+
+  @doc """
+  This function adds a manual to a bookmark and associates them
+  """
+  @spec add_manual_to_bookmark(Bookmark.t(), Manual.t()) :: {:ok, Bookmark.t()}|{:error, Ecto.Changeset.t()}
+  def add_manual_to_bookmark(%Bookmark{id: _id, manuals: manuals} = bookmark, %Manual{} = manual) do
+    bookmark
+    |> changeset(%{})
+    |> put_assoc(:manuals, manuals ++ [manual])
```

Comment:
appending to lists is very expensive since the compiler has to walk through all the items to add one perhaps consider prepending to the list like this `[manual] ++ manuals` see here https://elixirforum.com/t/when-would-you-need-to-append-to-an-elixir-list-over-prepending/12908